packages/client/src/create-deployment.ts

```diff
@@ -99,18 +99,33 @@ export default function buildCreateDeployment() {
             entries: fileList.map(file => relative(workPath, file)),
           })
           .pipe(createGzip());
-        const tarBuffer = await streamToBuffer(tarStream);
-        debug('Packed tarball');
-        files = new Map([
-          [
-            hash(tarBuffer),
-            {
-              names: [join(workPath, '.vercel/source.tgz')],
-              data: tarBuffer,
-              mode: 0o666,
-            },
-          ],
-        ]);
+        if (process.env.SPLIT_SOURCE_ARCHIVE) {
+          const chunkedTarBuffers = await streamToBufferChunks(tarStream);
+          debug(`Packed tarball into ${chunkedTarBuffers.length} chunks`);
+          files = new Map(
+            chunkedTarBuffers.map((chunk, index) => [
+              hash(chunk),
+              {
+                names: [join(workPath, `.vercel/source.tgz.part${index}`)],
```

Comment:
```suggestion
                names: [join(workPath, `.vercel/source.tgz.part${index+1}`)],
```

1-based indexing for the file names would be nice to match prior art: https://www.win-rar.com/split-files-archive.html?&L=0